src/components/users/edit-member-dialog.tsx

```diff
@@ -0,0 +1,321 @@
+          </MenuItem>)
+        }
+      </TextField>
+      <TextField
+        select
+        fullWidth
+        disabled={ !foundCardUsers.length }
+        size="medium"
+        onChange={ handleCardUserSelectChange }
+        label={ strings.userManagementScreen.editMemberDialog.cardAuthUsersSelectLabel }
+      >
+        { foundCardUsers.map(user => {
+          const federatedUser = user.federatedIdentities?.find(federatedIdentity => federatedIdentity.source === UserFederationSource.Card);
+          return (
+            <MenuItem key={ user.displayName} value={ federatedUser?.userId }>
+              { `${user.firstName} ${user.lastName}` }
+            </MenuItem>);
+        })
+        }
+      </TextField>
+      <FormControlLabel
+        control={
+          <Switch
+            checked={ linkSwitchChecked }
+            onChange={ handleSwitchChange }
+          />
+        }
+        label={ strings.userManagementScreen.editMemberDialog.userIsLinked }
+      />
+      <TextField
+        disabled={ !selectedMetaformUser ?? loading }
+        fullWidth
+        size="medium"
+        required={ true }
+        value={ selectedMetaformUser?.email ?? "" }
+        name="email"
+        type="email"
+        label={ strings.userManagementScreen.editMemberDialog.emailLabel }
+        onChange={ onTextFieldChange }
+      />
+      <TextField
+        disabled={ !selectedMetaformUser ?? loading }
+        fullWidth
+        size="medium"
+        required={ true }
+        value={ selectedMetaformUser?.firstName ?? "" }
+        name="firstName"
+        label={ strings.userManagementScreen.editMemberDialog.firstNameLabel }
+        onChange={ onTextFieldChange }
+      />
+      <TextField
+        disabled={ !selectedMetaformUser ?? loading }
+        fullWidth
+        size="medium"
+        required={ true }
+        value={ selectedMetaformUser?.lastName ?? "" }
+        name="lastName"
+        label={ strings.userManagementScreen.editMemberDialog.lastNameLabel }
+        onChange={ onTextFieldChange }
+      />
+      <TextField
+        disabled
+        fullWidth
+        size="medium"
+        label={ strings.userManagementScreen.editMemberDialog.upnNumberLabel }
+        value={ (selectedMetaformUser && getUpnNumber()) ?? "" }
+      />
+    </Stack>
+  );
+
+  /**
+   * Renders dialog actions
+   */
+  const renderDialogActions = () => [
+    <Button disableElevation variant="contained" onClick={ handleCancelClick } color="secondary" autoFocus>
+      { strings.userManagementScreen.editMemberDialog.cancelButton }
+    </Button>,
+    <GenericLoaderWrapper loading={ loading }>
+      <Button onClick={ handleEditClick } color="primary" disabled={ !valid }>
+        { strings.userManagementScreen.editMemberDialog.editButton }
+      </Button>
+    </GenericLoaderWrapper>
+  ];
+
+  useEffect(() => {
+    setLinkSwitchChecked(!!selectedMetaformUser?.federatedIdentities?.length ?? false);
+  }, [ selectedMetaformUser ]);
```

Comment:
Always place the useEffect hook call to the top of a component after the other hook calls and the method it calls.

Reply:
Resolved